### ISSUE TITLE
Bridgy Publish feature request: Support posting to GitLab sites

### ISSUE THREAD

**Issue by tantek** [snarfed/bridgy#1397]
It would be great if there was a way to post issues &amp; comments for GitLab repos, on a personal site, and have Bridgy Publish syndicate them to the actual GitLab repos. e.g. feature requests to Pleroma:

  * [https://git.pleroma.social/pleroma/pleroma/-/issues](https://git.pleroma.social/pleroma/pleroma/-/issues)

Such as:

  * Implement IndieAuth support (both sign-in with IndiAuth, and use of a Pleroma profile as an IndieAuth identity to sign-in elsewhere)
  * Implement Micropub server support
  * Implement Webmention support, sending &amp; receiving

(Originally published at: https://tantek.com/2023/019/b1/bridgy-publish-support-gitlab)

**Comment by snarfed** [comment]
Definitely! New silo instructions, for posterity: https://bridgy.readthedocs.io/#adding-a-new-silo

**Comment by snarfed** [comment]
APIs for [issues](https://docs.gitlab.com/ee/api/issues.html) and [comments](https://docs.gitlab.com/ee/api/notes.html) look pretty complete.

**Comment by jpcaruana** [comment]
If someone wanted to give this a try, would you recommand or caution against using the python client library https://python-gitlab.readthedocs.io/ ?

**Comment by jpcaruana** [comment]
> I'm now generally trying to avoid libraries and just use eg requests and requests-oauthlib directly.

I aggree with you.

> I noticed that GitLab has both REST and GraphQL APIs, like GitHub

Gitlab REST API is pretty simple to use. The only issue I can see here is: would we take into account self-hosted Gitlab instances?

**Comment by snarfed** [comment]
> The only issue I can see here is: would we take into account self-hosted Gitlab instances?

Maybe eventually, but I wouldn't worry about it for our first pass.
